spec/rocket_chat/messages/im_spec.rb

```diff
@@ -0,0 +1,149 @@
+              t: 'd',
+              msgs: 0,
+              ts: '2018-03-26T19:11:50.711Z',
+              meta: {
+                revision: 0,
+                created: 1522094603745,
+                version: 0
+              },
+              '$loki': 65,
+              usernames: [
+                'rocket.cat',
+                'user.test'
+              ]
+            },
+            success: true
+          }.to_json,
+          status: 200
+        )
+
+      stub_authed_request(:post, '/api/v1/im.create')
+        .with(
+          body: {
+            username: 'non-existent.user'
+          }
+        ).to_return(
+          body: { success: false }.to_json,
+          status: 400
+        )
+    end
+
+    context 'with a valid session' do
+      it 'create a new direct conversation' do
+        im = session.im.create username: 'rocket.cat'
+        expect(im).to be_a RocketChat::Room
+        expect(im.members).to eq %w[rocket.cat user.test]
+      end
+
+      it 'does not send attributes' do
+        expect do
+          session.im.create
+        end.to raise_error RocketChat::StatusError
+      end
+    end
+
+    context 'with an invalid session token' do
+      let(:token) { RocketChat::Token.new(authToken: nil, userId: nil) }
+
+      it 'raises a status error' do
+        expect do
+          session.im.create username: 'rocket.chat'
+        end.to raise_error RocketChat::StatusError, 'You must be logged in to do this.'
+      end
+    end
+
+    context 'when the user no exists' do
+      it 'raises a status error' do
+        expect do
+          session.im.create(username: 'non-existent.user')
+        end.to raise_error RocketChat::StatusError
+      end
+    end
+  end
+
+  describe '#counters' do
+    before do
+      stub_unauthed_request :get, '/api/v1/im.counters'
+
+      stub_authed_request(:get, '/api/v1/im.counters?roomId=rocket.cat&username=')
+        .to_return(
+          body: {
+            joined: true,
+            members: 2,
+            unreads: 0,
+            unreadsFrom: '2019-01-04T21:40:11.251Z',
+            msgs: 0,
+            latest: '2019-01-04T21:40:11.251Z',
+            userMentions: 0,
+            success: true
+          }.to_json,
+          status: 200
+        )
+
+      stub_authed_request(:get, '/api/v1/im.counters?roomId=1234&username=')
+        .to_return(
+          body: {
+            success: false,
+            error: '[invalid-channel]',
+            errorType: 'invalid-channel'
+          }.to_json,
+          status: 400
+        )
+
+      stub_authed_request(:get, '/api/v1/im.counters?roomId=&username=')
+        .to_return(
+          body: {
+            success: false,
+            error: 'Body param "roomId" or "username" is required [error-room-param-not-provided]',
+            errorType: 'error-room-param-not-provided'
+          }.to_json,
+          status: 400
+        )
+    end
+
+    context 'with a valid session' do
+      it 'get qty of messages' do
```

Comment:
Please don't use abbreviations in spec descriptions. They should be as readable as possible!